### PR TITLE
qlog/event.go: import the error structs from quic-go/internal/qerr

### PR DESCRIPTION
**Pull request by haxkor**
Instead of importing the error structs from the quic-go package, we can be more specific and import them from the qerr package.
If you tried to import the quic-go/qlog package you got an cyclic import error.
This fixes the problem.

### CODE REVIEW

**Comment by haxkor** [comment]
Hi, my apologies I really should have been more precise in the description of the problem:
If you add a new module that utilizes /quic/qlog, and go on to import this package in the /quic package, you get an import loop.
I have prepared [ a minimal example: ](https://github.com/haxkor/quic-go/commit/94ec85cd89f2503c6d15bd3848230dd05ca9bf79)

In my program utilizing my fork of quic, this is the go build output
`package myprogram
        imports github.com/quic-go/quic-go
        imports github.com/quic-go/quic-go/new_module
        imports github.com/quic-go/quic-go/qlog
        imports github.com/quic-go/quic-go: import cycle not allowed
`
This stems from /quic/qlog/events.go importing the error structs from /quic

**Comment by marten-seemann** [comment]
That’s a problem in your fork, not in quic-go. I don’t see why we should change quic-go in this case.

**Comment by haxkor** [comment]
Right now its not a problem in quic-go, true. It would become a problem once you want to add some functionality involving qlog.
In qlog/events.go, do you think it is better to import the error structs from quic instead of the more specific (original) location?

**Comment by marten-seemann** [comment]
> Right now its not a problem in quic-go, true. It would become a problem once you want to add some functionality involving qlog.

It really is not. qlog is working fine.

> In qlog/events.go, do you think it is better to import the error structs from quic instead of the more specific (original) location?

Yes. qlog is its own package, and there's no reason why it needs to depend on *anything* internal at all. It's just an implementation of the `logging.ConnectionTracer` interface.

**Comment by haxkor** [comment]
> It's just an implementation of the `logging.ConnectionTracer` interface.
Thank you for this, I had not thought of it this way. I wanted to reuse much of qlogs code but that was not possible without my changes.
I guess I will have to copy the relevant into my new package and end up with duplicates.

**Comment by marten-seemann** [comment]
Can you elaborate on the changes you want to make for qlog? Maybe that’s something that would be useful to add to the package in general?

**Comment by haxkor** [comment]
So the big thing I am working on is a mechanism that prioritizes bidirectional streams over unidirectional streams.
For that I need a tracer that records when a Bi/Uni frame is placed in the ringbuffer. Further, I also want information about congestion control. 

I do not want to change qlog, but reuse it.
Because the qlogs are still nice to have, I want to reuse many parts of the connectionTracer implementation to keep the qlog functionality. But maybe this should be solved by having a multiplexed tracer. I havent really gotten behind how I could implement that, maybe I should be looking into this.

What could not be solved with a multiplexed tracer is the reuse of the writer.go or event.go.

Right now I circumvent this issue by adding a function to qlog/connection_tracer.go that returns a connectionTracer.
See the code [here](https://github.com/haxkor/quic-go/blob/balancer/qlog/connection_tracer.go)
This way I will be able to reuse the qlog tracer but have a call to my extra functionality when I need it.
I also added 2 new event types (FrameBuffer Read / Write) to qlog/event.go

So in short I dont really want to improve qlog but reuse it. I am very new to quic-go so I appreciate your feedback on my way of doing things, please let me know if there is a better approach to solve this. 


Btw, as I looked at at qlog/event.go again, I realised qlog does already depend on some internal packages:
`	"github.com/quic-go/quic-go/internal/protocol"
	"github.com/quic-go/quic-go/internal/utils"

**Comment by marten-seemann** [comment]
I see, so you're forking the entire quic-go packet. That's definitely non-standard use of the library :)
You might be able to get around this with clever `replace` directives, but I'm not sure if that works in your use case.
That said, stream priorities is a very long-standing issue, see https://github.com/quic-go/quic-go/issues/437. Hopefully we can resolve at some point..

**Comment by haxkor** [comment]
Thank you for pointing me to the existing, longstanding discussion.
Do I understand correctly that there is no code enabling stream prioritisation?

My approach is quite hacky and distinguishes only between Uni and Bidi streams, so it will only be a POC.

>clever replace directives

Well I just implemented the fix proposed here:)
